frontend/public/components/default-resource.jsx

```diff
@@ -19,27 +19,58 @@ import {
 const { common } = Cog.factory;
 const menuActions = [...common];
 
-const Header = props => <ListHeader>
-  <ColHead {...props} className="col-xs-6 col-sm-4" sortField="metadata.name">Name</ColHead>
-  <ColHead {...props} className="col-xs-6 col-sm-4" sortField="metadata.namespace">Namespace</ColHead>
-  <ColHead {...props} className="col-sm-4 hidden-xs" sortField="metadata.creationTimestamp">Created</ColHead>
-</ListHeader>;
+const computeColumnSizes = (columns, isResourceNamespaced) => {
+  const columnsNumber = isResourceNamespaced ? _.size(columns) + 1 : _.size(columns);
+  const columnsSize = Math.floor(12 / columnsNumber);
+  const fistColumnSize = 12 - (columnsSize * (columnsNumber-1));
+  return {
+    fistColumnSize,
+    columnsSize
+  };
+};
+
+const CustomHeader_ = props => {
+  if (!_.get(props, ['resourceTable', 'table'])) {
+    return null;
+  }
+  const { table, isNamespaced } = props.resourceTable;
+  const { fistColumnSize, columnsSize } = computeColumnSizes(table.columnDefinitions, isNamespaced);
+  const heads = _.map(table.columnDefinitions, (column, index) => {
+    const size = index === 0 ? fistColumnSize : columnsSize;
+    return <ColHead key={_.uniqueId()} className={`col-xs-${size}`}>{column.name}</ColHead>;
+  });
+  isNamespaced && heads.splice(1, 0, (<ColHead key={_.uniqueId()} className={`col-xs-${columnsSize}`}>Namespace</ColHead>));
+  return <ListHeader>
+    {heads}
+  </ListHeader>;
+};
+
+const stateToProps = ({UI}) => {
+  const resourceTable = UI.get('resourceTable');
+  return {resourceTable};
+};
+
+const CustomHeader = connect(stateToProps)(CustomHeader_);
 
-const RowForKind = kind => function RowForKind_ ({obj}) {
+const RowForKind = (kind, rows, isNamespaced) => function RowForKind_ ({obj, index}) {
+  if (!rows) {
+    return null;
+  }
+  const rowCells = rows[index].cells;
```

Comment:
nit: Since you reference it more than once, I'd add a var for the row

```js
const row = rows[index];
const { cells } = row;
const namespace = _.get(row, 'object.metadata.namespace');
```